Assets/Mirror/Tests/Editor/WeaverAssembler.cs

```diff
@@ -1,14 +1,38 @@
+using System.Collections.Generic;
 using System.IO;
 using System.Linq;
-using System.Collections.Generic;
+using UnityEditor;
 using UnityEditor.Compilation;
 using UnityEngine;
 
 namespace Mirror.Tests
 {
     public class WeaverAssembler : MonoBehaviour
     {
-        public const string OutputDirectory = "Assets/Mirror/Tests/Editor/WeaverTests~/";
+        static string _outputDirectory;
+        public static string OutputDirectory
+        {
+            get
+            {
+                if (string.IsNullOrEmpty(_outputDirectory))
+                {
+                    string[] guidsFound = AssetDatabase.FindAssets($"t:Script " + nameof(WeaverAssembler));
+                    if (guidsFound.Length == 1 && !string.IsNullOrEmpty(guidsFound[0]))
+                    {
+                        string path = AssetDatabase.GUIDToAssetPath(guidsFound[0]);
```

Comment:
FindAssets, GUIDToAssetPath etc. - that's a lot of magic.
it's better if it works, but a test would probably be a good idea :)

Reply:
They are all calls to unity functions so should work all of the time. 

This is only used in the weaver tests so if something does go wrong here then the tests will fail with any error.